yb-voyager/src/queryissue/unsupported_dml_constructs.go

```diff
@@ -111,3 +111,93 @@ func (d *XmlExprDetector) Detect(msg protoreflect.Message) ([]string, error) {
 	}
 	return nil, nil
 }
+
+type RangeTableFunDetector struct{}
+
+func NewRangeTableFuncDetector() *RangeTableFunDetector {
+	return &RangeTableFunDetector{}
+}
+
+// Detect checks if a RangeTableFunc node is present for a XMLTABLE() function
+func (d *RangeTableFunDetector) Detect(msg protoreflect.Message) ([]string, error) {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_RANGETABLEFUNC_NODE {
+		if isXMLTable(msg) {
+			return []string{XML_FUNCTIONS}, nil
+		}
+	}
+	return nil, nil
+}
+
+/*
+XMLTABLE transforms XML data into relational table format, making it easier to query XML structures.
+Detection in RangeTableFunc Node:
+- docexpr: Refers to the XML data source, usually a column storing XML.
+- rowexpr: XPath expression (starting with '/' or '//') defining the rows in the XML.
+- columns: Specifies the data extraction from XML into relational columns.
+
+Example: Converting XML data about books into a table:
+SQL Query:
+
+	SELECT x.*
+	FROM XMLTABLE(
+		'/bookstore/book'
+		PASSING xml_column
+		COLUMNS
+			title TEXT PATH 'title',
+			author TEXT PATH 'author'
+	) AS x;
+
+Here, 'docexpr' points to 'xml_column' containing XML data, 'rowexpr' selects each 'book' node, and 'columns' extract 'title' and 'author' from each book.
+Hence Presence of XPath in 'rowexpr' and structured 'columns' typically indicates XMLTABLE usage.
+
+Function to detect if a RangeTableFunc node represents XMLTABLE()
+*/
+func isXMLTable(rangeTableFunc protoreflect.Message) bool {
+	log.Infof("checking if range table func node is for XMLTABLE()")
+	// Check for 'docexpr' field
+	docexprField := rangeTableFunc.Descriptor().Fields().ByName("docexpr")
+	if docexprField == nil {
+		return false
+	}
+	docexprNode := rangeTableFunc.Get(docexprField).Message()
+	if docexprNode == nil {
+		return false
+	}
+
+	// Check for 'rowexpr' field
+	rowexprField := rangeTableFunc.Descriptor().Fields().ByName("rowexpr")
+	if rowexprField == nil {
+		return false
+	}
+	rowexprNode := rangeTableFunc.Get(rowexprField).Message()
+	if rowexprNode == nil {
+		return false
+	}
+
+	xpath := queryparser.GetStringValueFromNode(rowexprNode)
+	log.Debugf("xpath expression in the node: %s\n", xpath)
+	// Keep both cases check(param placeholder and absolute check)
+	if xpath == "" {
+		// Attempt to check if 'rowexpr' is a parameter placeholder like '$1'
+		isPlaceholder := queryparser.IsParameterPlaceholder(rowexprNode)
+		if !isPlaceholder {
+			return false
+		}
+	} else if !queryparser.IsXPathExprForXmlTable(xpath) {
```

Comment:
Why do we need to validate the XPath? Won't it suffice to check that it is non-empty? We only want to detect if XMLTable function is being used, right? The fact that the query showed up in pg_stat_statements means that it will be a valid query, and therefore, a valid XPath?

Reply:
My understanding:
- RangeTableFunc node is a generic node, can be used apart from XMLTABLE() also
- It has three main components - rowexpr, docexpr, and columns
- These components can also be present for any other RangeTableFunc node, so we should also look at the content of components here. Hence check the value/str which we expect it to be xpath.


Also in most of the cases, pgss will replace with a placeholder like $1...
But it was not doing that in case of `CREATE VIEW... SELECT XMLTABLE()....` also the check should be fullproof so i separately handling if its placeholder then always true(chances of false positives are very low)

Jfyi, i couldn't find any other examples for RangeTableFunc node, apart from XMLTABLE(), but since it is a node, there are chances of more cases.

Reply:
Ah, and there's no way to figure out that directly that the `RangeTableFunc` node is an XMLTABLE node? That seems strange!

Reply:
Shooting in the dark here but from postgres code 
```static RangeTableFunc *
_copyRangeTableFunc(const RangeTableFunc *from)
{
	RangeTableFunc *newnode = makeNode(RangeTableFunc);

	COPY_SCALAR_FIELD(lateral);
	COPY_NODE_FIELD(docexpr);
	COPY_NODE_FIELD(rowexpr);
	COPY_NODE_FIELD(namespaces);
	COPY_NODE_FIELD(columns);
	COPY_NODE_FIELD(alias);
	COPY_LOCATION_FIELD(location);

	return newnode;
}
```

Could `alias` be helpful??

Reply:
hmm, code mentions `json_table` which is introduced in PG 17.
I think the only way is to check the xpath expression for xml table.

Also i am not sure if there are a lot of cases for table function apart from xmltable() and json_table()

Reply:
Jfyi, golang pg parser fails for parsing a SQL with json_table. 
SQL stmt: https://www.postgresql.org/docs/current/functions-json.html#:~:text=The%20following%20query%20shows%20how%20to%20use%20JSON_TABLE

Reply:
> hmm, code mentions json_table which is introduced in PG 17.

Pg-parser is on PG16 only yet. So It won't be able to detect further features

Reply:
I think if you look at the documentation in postgres - https://github.com/postgres/postgres/blob/ea792bfd93ab8ad4ef4e3d1a741b8595db143677/src/include/nodes/parsenodes.h#L651 

It says 
```
 * RangeTableFunc - raw form of "table functions" such as XMLTABLE
 *
 * Note: JSON_TABLE is also a "table function", but it uses JsonTable node,
 * not RangeTableFunc.
 ```
 
 So, as of now, XML_TABLE is the only case that uses RangeTableFunc. 
 Therefore, It's okay to go ahead with this for now IMO, but let's document in code this assumption  + link to postgres code above @sanyamsinghal

Reply:
oh ok, missed `but it uses JsonTable node`

Yeah makes sense, in future there can be more cases to consider under RangeTableFunc node but as not as of now.
Will do comments.

Reply:
done